### PR TITLE
Expand the smoke test with fmc-alias, rt-alias, and hitless update validation

### PR DESCRIPTION
**Pull request by korran**
Please review each commit separately.

### CODE REVIEW

File: test/tests/caliptra_integration_tests/smoke_test.rs [hunk 1]
```diff
@@ -395,6 +397,114 @@ fn smoke_test() {
...
+
+    let rt_dice_tcb_info = DiceTcbInfo::find_single_in_cert(rt_alias_cert_der).unwrap();
+    assert_eq!(
+        rt_dice_tcb_info,
+        Some(DiceTcbInfo {
+            vendor: Some("Caliptra".into()),
+            model: Some("RT".into()),
+            svn: Some(0x100),
+            fwids: vec![DiceFwid {
+                // FMC
```
Comment (jhand2) [inline test/tests/caliptra_integration_tests/smoke_test.rs:426]: nit: RT?

File: test/tests/caliptra_integration_tests/smoke_test.rs [hunk 2]
```diff
@@ -510,6 +510,147 @@ fn smoke_test() {
         .cptra_hw_error_non_fatal()
         .read()
         .mbox_ecc_unc());
+
+    let image2 = caliptra_builder::build_and_sign_image(
```
Comment (jhand2) [inline test/tests/caliptra_integration_tests/smoke_test.rs:514]: Maybe the ship has sailed, but the smoke test is a bit hard to follow since it does so many things. Would it make sense to factor the checks into helpers to make it a little easier to follow. Something like:

```
// Load firmware

verify_uart(...);
verify_cert_fields(...);

// Update firmware

verify_updated_cert_fields(...);
```

**Review by jhand2 — APPROVED** [review]

File: test/tests/caliptra_integration_tests/smoke_test.rs (hunk 2 above)
Comment (korran) [reply test/tests/caliptra_integration_tests/smoke_test.rs:514]: I do want to do a cleanup commit at some point, moving boilerplaty stuff like the golden-data checks to helper functions or macros. In general though, I think we want to keep important assertions like "the rt-alias must be signed by the fmc-alias" in the top-level test. It's more important that the assertions are clear and "obviously right at a glance". 

The next major change I want to make to the test is to parameterize the UDS and lifecycle/debug values.

Comment (jhand2) [reply test/tests/caliptra_integration_tests/smoke_test.rs:514]: Ack. Seems fine clean this up in a follow up. I agree having all the asserts in one place is helpful.

File: test/tests/caliptra_integration_tests/smoke_test.rs (hunk 1 above)
Comment (korran) [reply test/tests/caliptra_integration_tests/smoke_test.rs]: Done

**Review by jhand2 — APPROVED** [review]
